components/placesToVisitOverview/index.js

```diff
@@ -51,23 +29,31 @@ export default function PlacesToVisitOverview() {
       .catch((err) => console.log("Fetch Error :-S", err));
   }, []);
 
+  const toggleFavorite = (xid) => {
+    setPlaces(
+      places.map((place) => {
+        if (place.properties.xid === xid.properties.xid) {
+          return {
+            ...place,
+            isFavorite: !place.isFavorite,
+          };
+        }
+        return place;
+      })
+    );
+  };
+
   return (
     <div>
-      <h1>Places to Visit</h1>
+      <h1>Results:</h1>
       <ul>
         {places.map((place, index) => (
-          <li key={place.properties.xid}>
-            <span>{place.properties.name}</span>
-            <button
-              onClick={() => {
-                const newPlaces = [...places];
-                newPlaces[index].isFavorite = !newPlaces[index].isFavorite;
-                setPlaces(newPlaces);
-              }}
-            >
-              {place.isFavorite ? "★" : "☆"}
-            </button>
-          </li>
+          <Place
+            key={place.properties.xid}
+            name={place.properties.name}
+            onFavorite={toggleFavorite}
+            xid={place} // hier sollte xid={place.properties.xid} sein
```

Comment:
should this not be {place.properties.xid}?

Reply:
you are absolutely right. will change it